### PR TITLE
feat(dpp): random documents based on JSON schema

### PR DESCRIPTION
**Pull request by fominok**
## Issue being fixed or feature implemented

Adds a method under a feature flag for `DocumentType` to generate random documents using `json-schema-faker-rs` solution

## What was done?

Since jsonschema taken from `DocumentType` is not fully defined at one place as a self-contained schema, this PR wraps a json-schema-faker-rs part into manual tweaks to be able to go from json-schema-faker-rs output to an actual `Document`.

## How Has This Been Tested?

There is a test.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/Cargo.toml [hunk 1]
```diff
@@ -215,3 +217,4 @@ extended-document = [
 
 factories = []
 client = ["factories", "state-transitions"]
+documents-faker = ["dep:json-schema-faker", "document-serde-conversion"]
```
Comment (shumkov) [inline packages/rs-dpp/Cargo.toml:220]: Please add a comment about additional extra dependencies

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/Cargo.toml [hunk 2]
```diff
@@ -74,6 +72,8 @@ tokio = { version = "1", features = [
 ] }
 tokio-util = { version = "0.7.8" }
 derive_more = "0.99.17"
+metrics = "0.22.1"
```
Comment (shumkov) [inline packages/rs-drive-abci/Cargo.toml:75]: Please revert unrelated changes

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/Cargo.toml (hunk 2 above)
Comment (fominok) [reply packages/rs-drive-abci/Cargo.toml:75]: this change is related: previous version of `metrics` indirectly had version of another library that went in conflict with one used in faker dependency

Comment (fominok) [reply packages/rs-drive-abci/Cargo.toml]: nvm it seems dev was updated in these months

**Review by shumkov — DISMISSED** [review]
👍

File: packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs [hunk 3]
```diff
@@ -21,11 +30,127 @@ use crate::identity::Identity;
...
+            extended_document_property_names::FEATURE_VERSION,
+            property_names::{ID, OWNER_ID, REVISION},
+            serialization_traits::DocumentPlatformValueMethodsV0,
+        };
+
+        let json_schema = &self
+            .schema
+            .clone()
+            .try_into()
+            .expect("jsonschema must be a valid JSON");
```
Comment (shumkov) [inline packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs:57]: Return an error. Schema might not be valid if you create a contract without `full_validation` flag

File: packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs [hunk 4]
```diff
@@ -21,11 +30,123 @@ use crate::identity::Identity;
...
+        use anyhow::Context;
+
+        use crate::document::{
+            extended_document_property_names::FEATURE_VERSION,
+            property_names::{ID, OWNER_ID, REVISION},
+            serialization_traits::DocumentPlatformValueMethodsV0,
+        };
+
+        let json_schema = &self.schema.clone().try_into()?;
+        let json_documents = json_schema_faker::generate(&json_schema, count as u16)
```
Comment (shumkov) [inline packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs:54]: Clippy warning

File: packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs [hunk 5]
```diff
@@ -21,11 +30,123 @@ use crate::identity::Identity;
...
+                                        Value::U64(x) => Value::I64(x as i64),
+                                        Value::U32(x) => Value::I64(x as i64),
+                                        Value::I32(x) => Value::I64(x as i64),
+                                        x => x,
+                                    };
+                                })
+                        });
+                    }
+                    _ => {}
+                }
```
Comment (shumkov) [inline packages/rs-dpp/src/data_contract/document_type/v0/random_document.rs:144]: Clippy warning

**Review by shumkov — CHANGES_REQUESTED** [review]
